sys/include/riotboot/flashwrite.h

```diff
@@ -67,7 +106,16 @@ typedef struct {
     int target_slot;                        /**< update targets this slot     */
     size_t offset;                          /**< update is at this position   */
     unsigned flashpage;                     /**< update is at this flashpage  */
-    uint8_t flashpage_buf[FLASHPAGE_SIZE];  /**< flash writing buffer         */
+
+    /**
+     * @brief flash writing buffer
+     */
+    uint8_t RIOTBOOT_FLASHPAGE_BUFFER_ATTRS
+        flashpage_buf[RIOTBOOT_FLASHPAGE_BUFFER_SIZE];
+#if CONFIG_RIOTBOOT_FLASHWRITE_RAW
+    uint8_t RIOTBOOT_FLASHPAGE_BUFFER_ATTRS
+        firstblock_buf[RIOTBOOT_FLASHPAGE_BUFFER_SIZE];
+#endif
```

Comment:
The second buffer buffers the first few bytes of the flash update. This is required because some devices have a raw block size of multiple words, so we can't simply insert the checksum in the finish stage, but have to insert the whole raw block containing the checksum. I'll add something to the doxygen